src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -1814,7 +1814,7 @@ static void window_ride_construction_construct(rct_window* w)
         _currentTrackSelectionFlags |= TRACK_SELECTION_FLAG_TRACK_PLACE_ACTION_QUEUED;
     }
 
-    if (gTrackGroundFlags & TRACK_ELEMENT_LOCATION_IS_UNDERGROUND)
+    if (res->GroundFlagCount & TRACK_ELEMENT_LOCATION_IS_UNDERGROUND)
```

Comment:
Ah sorry forgot this wasn't inside the callback. You will have to do a cast to the correct type. `dynamic_cast<TrackPlaceActionResult*>(res)->GroundFlagCount` will compile.

Reply:
Ah no that wont work as this is a smart pointer.

Reply:
`dynamic_cast<TrackPlaceActionResult*>(res.get())->GroundFlagCount` now that will compile.